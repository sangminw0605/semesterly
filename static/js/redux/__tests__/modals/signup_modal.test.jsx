/*
Copyright (C) 2017 Semester.ly Technologies, LLC

Semester.ly is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

Semester.ly is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
*/
import React from 'react';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import Provider from 'react-redux/src/components/Provider';
import configureMockStore from 'redux-mock-store';
import SignupModalContainer from '../../ui/containers/modals/signup_modal_container';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Signup Modal', () => {
  it('shows when isVisible is true', () => {
    const store = mockStore({
      signupModal: { isVisible: true },
    });

    const { container } = render(
      <Provider store={store}>
        <SignupModalContainer />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('is hidden when isVisible is false', () => {
    const store = mockStore({
      signupModal: { isVisible: false },
    });
    const { container } = render(
      <Provider store={store}>
        <SignupModalContainer />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
