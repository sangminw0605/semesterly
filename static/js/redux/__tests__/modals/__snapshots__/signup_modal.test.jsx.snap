// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Modal is hidden when isVisible is false 1`] = `<div />`;

exports[`Signup Modal shows when isVisible is true 1`] = `
<div>
  <span>
    <div
      style="z-index: 1050; position: fixed; width: 100%; top: 50%; left: 50%;"
      class="signup-modal max-modal"
    >
      <div
        tabindex="-1"
        style="margin: 0px; background-color: white;"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="pro-pic"
              style="background-image: url(/static/img/blank.jpg);"
            />
            <h1>
              That feature requires an account...
            </h1>
          </div>
        </div>
        <div
          class="features"
        >
          <div
            class="feature-highlight"
          >
            <div
              class="row"
            >
              <div
                class="col-1-2"
              >
                <div
                  class="emoji"
                >
                  <i
                    class="fa fa-check"
                  />
                </div>
                Find classes with friends
                <img
                  class="sample-slot"
                  alt=""
                  src="/static/img/sample_slot.png"
                />
              </div>
              <div
                class="col-1-2"
              >
                <div
                  class="emoji"
                >
                  <i
                    class="fa fa-check"
                  />
                </div>
                Save & name multiple timetables
                <img
                  class="sample-slot"
                  alt=""
                  src="/static/img/multi_tt_sample.png"
                />
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-1-2"
              >
                <div
                  class="emoji"
                >
                  <i
                    class="fa fa-check"
                  />
                </div>
                Create custom events
                <img
                  alt=""
                  class="sample-slot"
                  src="/static/img/sample_custom_slot_grey.png"
                />
              </div>
              <div
                class="col-1-2"
              >
                <div
                  class="emoji"
                >
                  <img
                    class="emoji"
                    draggable="false"
                    alt="ðŸ”¥"
                    src="https://twemoji.maxcdn.com/2/72x72/1f525.png"
                  />
                </div>
                It's all free
                <h1>
                  More Burritos!
                </h1>
              </div>
            </div>
          </div>
          <div
            class="call-to-action"
          >
            <div
              class="disclaimer"
            >
              Semester.ly will NEVER post to your timeline. Your course selections will not be shared with any other user without your permission.
            </div>
            <a
              href="/login/facebook/"
            >
              <div
                class="signup-button"
              >
                Signup!
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      style="position: fixed; top: 0px; right: 0px; bottom: 0px; left: 0px; z-index: 1040; background-color: rgb(55, 58, 71);"
    />
  </span>
</div>
`;
